src/token.rs

```diff
@@ -32,19 +36,58 @@ pub enum Request {
     },
 }
 
-#[derive(Debug, Default, Deserialize, Serialize)]
-pub struct ExtraResponseTokenFields {
-    #[serde(skip_serializing_if = "Option::is_none")]
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ExtraResponseTokenFields<AD>
+where
+    AD: AuthorizationDetailsProfile,
+{
     pub c_nonce: Option<Nonce>,
-    #[serde(skip_serializing_if = "Option::is_none")]
     pub c_nonce_expires_in: Option<Duration>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub authorization_pending: Option<bool>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub interval: Option<Duration>,
+    #[serde(bound = "AD: AuthorizationDetailsProfile")]
+    pub authorization_details: Option<Vec<AuthorizationDetail<AD>>>,
+}
+
+impl<AD> Default for ExtraResponseTokenFields<AD>
```

Comment:
It's not able to derive `Default` automatically because of the generic?

Reply:
I saw an error at some point and changed to a manual implementation, but after your comment I went and tested it again with the derive and it worked fine. I most likely misinterpreted some unrelated error.